libs/sdk-ui-vis-commons/src/legend/StaticLegend.tsx

```diff
@@ -78,21 +92,38 @@ export class StaticLegend extends React.PureComponent<IStaticLegendProps> {
             );
         }
 
-        const seriesCount = series.length;
-        const { hasPaging, visibleItemsCount } = calculateStaticLegend(seriesCount, containerHeight);
+        const columnNum = position === "dialog" ? 2 : 1;
+
+        const labelHeight = label ? ITEM_HEIGHT : 0;
+        const labelComponent = label ? <LegendLabelItem label={label} /> : null;
+        const contentHeight = containerHeight - labelHeight;
 
+        const seriesCount = series.length;
+        const { hasPaging, visibleItemsCount } = calculateStaticLegend(
+            seriesCount,
+            contentHeight,
+            columnNum,
+            paginationHeight,
+        );
         const start = (page - 1) * visibleItemsCount;
         const end = Math.min(visibleItemsCount * page, series.length);
 
         const pagedSeries = series.slice(start, end);
 
-        const heightOfAvailableSpace = visibleItemsCount * ITEM_HEIGHT;
-        const heightOfVisibleItems = Math.min(visibleItemsCount, seriesCount) * ITEM_HEIGHT;
-        const seriesHeight = shouldFillAvailableSpace ? heightOfAvailableSpace : heightOfVisibleItems;
+        const heightOfAvailableSpace = (visibleItemsCount / columnNum) * ITEM_HEIGHT;
+        const heightOfVisibleItems = Math.min(visibleItemsCount / columnNum, seriesCount) * ITEM_HEIGHT;
+        const seriesHeight =
+            (shouldFillAvailableSpace ? heightOfAvailableSpace : heightOfVisibleItems) + labelHeight;
+        const visibleItemsFitOneColumn = shouldItemsFitOneColumn(
+            visibleItemsCount,
+            columnNum,
+            pagedSeries.length,
+        );
 
         return (
-            <div className={classNames}>
+            <div className={`${classNames} ${visibleItemsFitOneColumn}`}>
```

Comment:
Please use the `classnames` package to combine multiple class names.

Reply:
fixed by 650ddd8

Reply:
There is still a template string in the div `className` prop, this should also be a call to classnames

Reply:
ah, forgot to remove the second part :-) 75edb78 should be it :-)

Reply:
Fixed